### PR TITLE
Post /api/articles/ article id/comments

### PR DESCRIPTION
**Pull request by Mburnand-tech**
Not sure exactly what to say to help review but:

Implemented POST request, 

Testing is in separate Describe block below

Function created is 'newComment' in controller & 'postComment' in models.

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -89,3 +89,51 @@ describe('GET requests', () => {
...
+    test('/api/articles/:article_id/comments: should return the posted comment', () => {
+        return request(app).post('/api/articles/9/comments')
+        .send({
+            username : 'icellusedkars',
+            body : 'What an interesting post ... Not! Test body'
+        })
+        .expect(201)
+        .then(( {body}) => {
+            
+            //console.log(body[0])
```
Comment (itsaliabdallah) [inline __tests__/app.test.js:102]: remember to remove console logs

File: __tests__/app.test.js [hunk 2]
```diff
@@ -89,3 +89,51 @@ describe('GET requests', () => {
...
+        .send({
+            username : 'icellusedkars',
+            body : 'What an interesting post ... Not! Test body'
+        })
+        .expect(400)
+        .then(({body}) => {
+            expect(body.code).toBe('22P02')
+        })
+    });
+    test('/api/articles/9/comments: should return error for missing username', () => {
```
Comment (itsaliabdallah) [inline __tests__/app.test.js:127]: I'd seperate these two tests out as currently the article ID doesn't exist either and you're testing the username doesn't exist also in one test.

File: __tests__/app.test.js [hunk 3]
```diff
@@ -89,3 +89,51 @@ describe('GET requests', () => {
...
+            username : 'NotaUsername',
+            body : 'What an interesting post ... Not! Test body'
+        })
+        .expect(404)
+        .then(({body}) => {
+            expect(body.msg).toBe('Resource does not exist')
+        })
+    })
+});
+
```
Comment (itsaliabdallah) [inline __tests__/app.test.js:139]: I'd add another test what if either the username or the body doesn't exist on the object or are empty

File: models/news.models.js [hunk 4]
```diff
@@ -37,12 +37,22 @@ const findArticle = (article_id) => {
     })
 }
 
-
+const postComment = (comment, article_id) => {
+
+    return db.query(`INSERT INTO comments
+                    (body, author, article_id, votes, created_at)
```
Comment (itsaliabdallah) [inline models/news.models.js:43]: You don't need to pass votes and created_at into the INSERT INTO as the database creates these.

File: app/news.app.js [hunk 5]
```diff
@@ -2,14 +2,16 @@ const { response, request } = require('express')
...
 
 app.get('/api/topics', newsTopics)
 app.get('/api/articles', newsArticles)
 app.get(`/api/articles/:article_id`, specficNewsArticle)
 
+app.post('/api/articles/:article_id/comments', newComment)
+
 
 
 app.use((err, request, response, next) => {
```
Comment (itsaliabdallah) [inline app/news.app.js:17]: When it comes to your error handlers make sure you're sending some sort of message back to as these will what will be usefull to the end user on your frontend

**Review by itsaliabdallah — CHANGES_REQUESTED** [review]
Please see comments and add the correct testing and then the correct error handling for that test and resubmit!
